src/Code.gs

```diff
@@ -433,151 +424,157 @@ const ChatGPTApp = (function () {
         if (vision) {
           // Avoid hallucination & duplicated image description 
           if (!messages[messages.length - 1].content.includes("An image was given, here is the description:")) {
-            functions.push(imageDescriptionFunction);
+            tools.push({
+              type: "function",
+              function: imageDescriptionFunction
+            });
           }
         }
 
-        if (googleSheetAccess) {
-          functions.push(getDataFromGoogleSheetsFunction);
-        }
-
-        if (functions.length >> 0) {
+        if (tools.length >> 0) {
           // the user has added functions, enable function calling
+
+
           functionCalling = true;
-          let payloadFunctions = Object.keys(functions).map(f => ({
-            name: functions[f].toJSON().name,
-            description: functions[f].toJSON().description,
-            parameters: functions[f].toJSON().parameters
+          let payloadTools = Object.keys(tools).map(t => ({
+            type: "function",
+            function: {
+              name: tools[t].function.toJSON().name,
+              description: tools[t].function.toJSON().description,
+              parameters: tools[t].function.toJSON().parameters
+            }
           }));
-          payload.functions = payloadFunctions;
+          payload.tools = payloadTools;
 
-          if (!payload.function_call) {
-            payload.function_call = 'auto';
+          if (!payload.tool_choice) {
+            payload.tool_choice = 'auto';
           }
 
-          if (advancedParametersObject?.function_call &&
-            JSON.stringify(payload.function_call) !== '{"name":"urlFetch"}' &&
-            JSON.stringify(payload.function_call) !== '{"name":"webSearch"}') {
+          if (advancedParametersObject?.tool_choice &&
```

Comment:
It was already there before, but isn't this too complicated for nothing?

Reply:
eg instead of 
JSON.stringify(payload.tool_choice) !== '{"name":"urlFetch"}'
can't we do something like 
payload.tool_choice.name != "urlFetch"